youtube_dl/extractor/rtvs.py

```diff
@@ -36,12 +43,66 @@ class RTVSIE(InfoExtractor):
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
-        webpage = self._download_webpage(url, video_id)
+        if url.find('/radio/') != -1:
+            a2 = url.split('/')[-1]
+            a1 = url.split('/')[-2]
+            embed = self._download_webpage(
+                "https://www.rtvs.sk/embed/radio/archive/%s/%s" % (a1, a2),
+                video_id)
+            audio_id = re.search('audio5f.json?id=(?P<id>[^\"]+)', embed)
+            audio_id = audio_id.group('id')
+            info = self._download_json(
+                "https://www.rtvs.sk/json/audio5f.json?id=%s" % audio_id,
+                audio_id)
 
-        playlist_url = self._search_regex(
-            r'playlist["\']?\s*:\s*(["\'])(?P<url>(?:(?!\1).)+)\1', webpage,
-            'playlist url', group='url')
+            formats = []
+            formats.append({
+                'url': info['playlist'][0]['sources'][0]['src'],
+                'format_id':  None,
+                'height': 0
+                })
+            info = info['playlist'][0]
+            return {
+                'id': audio_id,
+                'title': info.get('title'),
+                'thumbnail': info.get('image'),
+                'formats': formats
+            }
+        else:
+            info = self._download_json(
+                "https://www.rtvs.sk/json/archive5f.json?id=%s" % video_id,
+                video_id)
+            info = info.get('clip')
 
-        data = self._download_json(
-            playlist_url, video_id, 'Downloading playlist')[0]
-        return self._parse_jwplayer_data(data, video_id=video_id)
+            formats = []
```

Comment:
Is this the same code as for Markiza? If so, are we looking at some well known framework? In that case we should consider creating a `_parse_whatever_data()` method of `InfoExtractor` that can be used elsewhere.